packages/web3wallet/src/controllers/engine.ts

```diff
@@ -0,0 +1,119 @@
+import { AuthClient, AuthEngineTypes, IAuthClient } from "@walletconnect/auth-client";
+import { SignClient } from "@walletconnect/sign-client";
+import { ISignClient } from "@walletconnect/types";
+import { IWeb3WalletEngine, Web3WalletTypes } from "../types";
+
+export class Engine extends IWeb3WalletEngine {
+  public signClient: ISignClient;
+  public authClient: IAuthClient;
+
+  constructor(client: IWeb3WalletEngine["client"]) {
+    super(client);
+    // initilized in init()
+    this.signClient = {} as any;
+    this.authClient = {} as any;
+  }
+
+  public init = async () => {
+    this.signClient = await SignClient.init({
+      core: this.client.core,
+      projectId: "",
```

Comment:
I couldn't find where project id is set for these latter on, should this come from some sort of options as opposed to being a string?

Reply:
`projectId` is set in the `core` instance that is passed in the `Web3Wallet` like so
https://github.com/WalletConnect/walletconnect-monorepo/blob/feat/web3wallet/packages/web3wallet/test/sign.spec.ts#L32

Reply:
Can't we simply omit `projectId` if it's already present via core? i.e. [SignClient.Options](https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/types/src/sign-client/client.ts#L46)

Reply:
removing it in `sign` but it's required param in `auth` :+1: